basis/listener/listener.factor

```diff
@@ -24,8 +24,13 @@ H{
 M: object prompt.
     nip prompt-style get-global format bl flush ;
 
+: with-ctrl-break ( quot -- )
+    enable-ctrl-break
+    [ disable-ctrl-break ] [ ] cleanup ; inline
+
 : parse-lines-interactive ( lines -- quot/f )
-    [ parse-lines ] with-compilation-unit ;
+    [ [ parse-lines ] with-ctrl-break
```

Comment:
So, we were waiting until 0.98 is released to merge this, mostly because of this change right here that enables it... instead of being optional before 0.98.  So, we could either make it optional here, and disabled by default, and merge before 0.98 (which is waiting for a couple fixes and @erg and I have to have some time), or wait until afterward to merge.